src/rule_definition_tools.jl

```diff
@@ -208,9 +209,19 @@ end
 function propagation_expr(Δs, ∂s)
     # This is basically Δs ⋅ ∂s
     ∂s = map(esc, ∂s)
+    n∂s = length(∂s)
 
-    ∂_mul_Δs = ntuple(i->:($(∂s[i]) * $(Δs[i])), length(∂s))
-    return :(+($(∂_mul_Δs...)))
+    ∂_mul_Δs = ntuple(i->:($(∂s[i]) * $(Δs[i])), n∂s)
+
+    # avoiding the extra `+` operation, it is potentially
+    # expensive for vector mode AD
+    sumed_∂_mul_Δs = if n∂s > 1
+        :(@. +($(∂_mul_Δs...)))
```

Comment:
Why use the macro?
`.+($(...))` 
should be identical

Reply:
Because you want to fuse https://github.com/JuliaDiff/ChainRulesCore.jl/pull/93/files/65b2ce27b789af574f6f1952d23b2174cd6bcf1c#diff-920b9886862d1cc24474b87677098488R214

Reply:
And it is more readable.

Reply:
I find it less readable.
And this is kind of why -- it's not as obvious what is being broadcast.
E.g. This case where we are also broadcasting the stuff from the other line.

I'ld just make both explicit broadcasts.

--


I know the DiffEq code base uses this a ton.
I have never used it except during the transition from implict broadcast back in 0.5ish

Reply:
```suggestion
        :(.+($(∂_mul_Δs...)))
```

Reply:
Oh, now I remember. Using explicit broadcast broke tests on older versions of Julia.

Reply:
It looks right to me when I do `macroexpand` but indeed it doesn't work in 1.0 for me locally either.

Reply:
```suggestion
        # we use `@.`  to broadcast the . and +
        # Note: we don't want to do broadcasting to occur if only 1 multiply (no +), 
        # because some arrays overload multiply with scalar, and those  that don't fall back to broadcasting anyway.
        # Note also: due to bugs in Julia 1.0 can't use `.+`  or `.*` inside expression literals anyway
        :(@. +($(∂_mul_Δs...)))
```

Reply:
Lets leave a comment so we remember why, then this can be merged.